eng/common/pipelines/templates/jobs/archetype-spec-gen-sdk.yml

```diff
@@ -168,6 +168,34 @@ jobs:
         artifactName: $(sdkArtifactName)
         targetPath: "$(System.DefaultWorkingDirectory)/generatedSdkArtifacts"
 
+    - task: PowerShell@2
+      displayName: Add label to the spec PR
+      condition: and(eq(variables['Build.Reason'], 'PullRequest'), ne(variables['BreakingChangeLabel'], ''), eq(variables['BreakingChangeLabelAction'], 'add'))
+      inputs:
+        pwsh: true
+        workingDirectory: $(SdkRepoDirectory)
+        filePath: $(SdkRepoDirectory)/eng/common/scripts/Add-IssueLabels.ps1
+        arguments: >
+          -RepoOwner $(SpecRepoOwner)
+          -RepoName $(SpecRepoName)
+          -IssueNumber "$(System.PullRequest.PullRequestNumber)"
+          -Labels $(BreakingChangeLabel)
+          -AuthToken "$(azuresdk-github-pat)"
+
+    - task: PowerShell@2
+      displayName: Remove label from the spec PR
+      condition: and(eq(variables['Build.Reason'], 'PullRequest'), ne(variables['BreakingChangeLabel'], ''), eq(variables['BreakingChangeLabelAction'], 'remove'))
+      inputs:
+        pwsh: true
+        workingDirectory: $(SdkRepoDirectory)
+        filePath: $(SdkRepoDirectory)/eng/common/scripts/Remove-IssueLabel.ps1
```

Comment:
It seems simpler to have one script that can process the `BreakingChangeLabelAction` value (add, remove, or ignore other values). That way we don't have to have multiple yaml steps with different conditionals.

Reply:
You are right, I considered that way as it does simplify the yml part. However, it would require the script to handle all different cases. I finally ended up with this approach, balancing the readability and maintainability.